### PR TITLE
Replace modals with PM, remove /warning

### PR DESCRIPTION
**Pull request by DrAnax**
- Replace modals when the user types a command with PM.- 
- Remove `/warning` (probably made for debug purposes)- 
- Add chat.addSelf message types.- 
- Replace "that command" with specific commands.

### CODE REVIEW

File: packages/client/src/chat.ts [hunk 1]
```diff
@@ -589,16 +589,30 @@ export function add(data: ChatMessage, fast: boolean): void {
 }
 
 // addSelf is used when the client needs to send a chat message to itself
-export function addSelf(msg: string, room: string): void {
+export function addSelf(
+  msg: string,
+  room: string,
+  type: "normal" | "info" | "error" = "normal",
```
Comment (Zamiell) [inline packages/client/src/chat.ts:595]: use an enum

File: packages/client/src/chat.ts [hunk 2]
```diff
@@ -589,16 +589,30 @@ export function add(data: ChatMessage, fast: boolean): void {
...
+): void {
+  let message = msg;
```
Comment (Zamiell) [inline packages/client/src/chat.ts:597]: make a getMessage function

File: packages/client/src/chat.ts [hunk 3]
```diff
@@ -12,6 +12,12 @@ import { parseIntSafe } from "./misc";
 import * as modals from "./modals";
 import ChatMessage from "./types/ChatMessage";
 
+export enum SelfChatMessageType {
+  normal,
```
Comment (Zamiell) [inline packages/client/src/chat.ts:16]: in typescript, the values of an enum should be capitalized, see: https://www.typescriptlang.org/docs/handbook/enums.html

File: packages/client/src/chat.ts [hunk 4]
```diff
@@ -589,16 +595,21 @@ export function add(data: ChatMessage, fast: boolean): void {
 }
 
 // addSelf is used when the client needs to send a chat message to itself
-export function addSelf(msg: string, room: string): void {
+export function addSelf(
+  msg: string,
+  room: string,
+  type: SelfChatMessageType = SelfChatMessageType.normal,
```
Comment (Zamiell) [inline packages/client/src/chat.ts:601]: the type annotation is superfluous

File: packages/client/src/chat.ts [hunk 5]
```diff
@@ -589,16 +595,21 @@ export function add(data: ChatMessage, fast: boolean): void {
...
+): void {
+  const message = getChatMessage(msg, type);
   add(
     {
-      msg,
-      who: "",
+      msg: message,
+      who: "Hanab Live",
```
Comment (Zamiell) [inline packages/client/src/chat.ts:607]: my bet is that this string is a constant that lives somewhere else, and if it is, then it shouldn't be hardcoded here

File: packages/client/src/chat.ts [hunk 6]
```diff
@@ -699,3 +710,17 @@ export function updatePeopleTyping(): void {
   chat1.html(msg);
   chat2.html(msg);
 }
+
+function getChatMessage(msg: string, type: SelfChatMessageType): string {
+  let message = msg;
+  switch (type) {
+    case SelfChatMessageType.info:
```
Comment (Zamiell) [inline packages/client/src/chat.ts:717]: please use { } brackets around the cases of all switch statements to match the existing code style

File: packages/client/src/chat.ts [hunk 7]
```diff
@@ -699,3 +710,17 @@ export function updatePeopleTyping(): void {
...
+      message = `<span class="green">${message}</span>`;
+      break;
+    case SelfChatMessageType.error:
+      message = `<span class="red">${message}</span>`;
+      break;
+    default:
```
Comment (Zamiell) [inline packages/client/src/chat.ts:723]: use ensure all cases

File: packages/client/src/lobby/createReplayJSON.ts [hunk 8]
```diff
@@ -63,9 +64,10 @@ export default function createJSONFromReplay(room: string) {
     () => {},
```
Comment (Zamiell) [inline packages/client/src/lobby/createReplayJSON.ts:64]: get rid of the then crap

File: packages/client/src/chat.ts (hunk 5 above)
Comment (DrAnax) [reply packages/client/src/chat.ts:607]: it's unreachable (`server_old/src/...` and `constants.go`), putting it into /data

File: packages/client/src/lobby/createReplayJSON.ts (hunk 8 above)
Comment (DrAnax) [reply packages/client/src/lobby/createReplayJSON.ts:64]: need to accept either this or #2574 first so I known where to merge from

Comment (Zamiell) [reply packages/client/src/lobby/createReplayJSON.ts:64]: up to you, which one do you want to focus on first

File: packages/client/src/chat.ts (hunk 7 above)
Comment (DrAnax) [reply packages/client/src/chat.ts:723]: in default there's no message change but code could reach that point; should I add a simple break?

Comment (Zamiell) [reply packages/client/src/chat.ts:723]: ill add it

File: packages/client/src/lobby/createReplayJSON.ts (hunk 8 above)
Comment (DrAnax) [reply packages/client/src/lobby/createReplayJSON.ts:64]: accept the other one first please.

File: packages/client/src/chat.ts (hunk 7 above)
Comment (Zamiell) [reply packages/client/src/chat.ts:723]: the point of the function is to return values, not modify variables, get rid of the let

**Comment by Zamiell** [comment]
thanks

**Pull request merged**